src/librustc_resolve/lib.rs

```diff
@@ -2296,12 +2340,32 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
                                           ty.span,
                                           ResolutionError::SelfUsedOutsideImplOrTrait);
                         } else {
+                            let segment = path.segments.last();
+                            let segment = segment.expect("missing name in path");
+                            let name = segment.identifier.name;
+
+                            let candidates: Vec<_> =
+                                self.lookup_suggestions(
+                                    name,
+                                    LookupTarget::AllCrates,
+                                    Namespace::TypeNS,
+                                    move |def: Def|
+                                        match def {
```

Comment:
Nit: I'd move the `match` up to the previous line, to save on some indentation, and drop the `move`